core/trino-main/src/main/java/io/trino/server/protocol/Query.java

```diff
@@ -463,18 +463,24 @@ private synchronized QueryResults getNextResult(long token, UriInfo uriInfo, Dat
         QueryInfo queryInfo = queryManager.getFullQueryInfo(queryId);
         queryManager.recordHeartbeat(queryId);
 
-        closeExchangeIfNecessary(queryInfo);
-
-        // fetch result data from exchange
-        QueryResultRows resultRows = removePagesFromExchange(queryInfo, targetResultSize.toBytes());
+        boolean isStarted = queryInfo.getState().ordinal() > QueryState.STARTING.ordinal();
```

Comment:
Is this change unrelated to the `submitted` change above?  If so, can you move this to another commit and explain why the change is needed (I don't understand why this change is being made)

Reply:
This is related. Without this, when `getQueryResult` is called during planning, it will call `closeExchangeIfNecessary` and `removePagesFromExchange`, and then the query will get stuck. Previously, this function was not called because `getQueryResult` returned a dummy result through `QueuedStatementResource`.